### DISCUSSION TITLE
Error in diagnosis related to internet access problems

### DISCUSSION THREAD

**Discussion by MarioExperience** [dani-garcia/vaultwarden#5273] [Q&A]
### Vaultwarden Support String

```
Versions
Server Installed
1.32.5
Server Latest Unknown
-
Web Installed
2024.6.2c
Database
SQLite: 3.46.0
Checks
OS/Arch
linux / x86_64
Running within a container
Yes (Base: Debian)
Environment settings overridden
No
Uses a reverse proxy
Yes
IP header Match
Config/Server: X-Real-IP
Internet access Error
No
Internet access via a proxy
Yes
DNS (github.com) Ok
140.82.121.3
Date & Time (Local)
Server: 2024-12-10 09:48:19 +01:00
Date & Time (UTC) Server/Browser Ok
NTP: Unable to fetch NTP time.Server: 2024-12-10 08:48:19 UTCBrowser: 2024-12-10 08:48:21 UTC
Domain configuration Match HTTPS
Server: https://problemsvaultwarden/admin/diagnosticsBrowser: https://problemsvaultwarden/admin/diagnostics
```

### Vaultwarden Build Version

1.32.5

### Deployment method

Official Container Image

### Reverse Proxy

nginx 1.27.0

### Host/Server Operating System

Linux

### Clients

Web Vault

### Steps To Reproduce

N/A

### Expected Result

In the "Internet Access" section it should appear as Yes, because the proxy is configured and it is observed that it has Internet access:

```
root@bitwarden:/# apt update
Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
Get:4 http://deb.debian.org/debian bookworm/main amd64 Packages [8789 kB]
Get:5 http://deb.debian.org/debian bookworm-updates/main amd64 Packages [8856 B]
Get:6 http://deb.debian.org/debian-security bookworm-security/main amd64 Packages [214 kB]
Fetched 9266 kB in 2s (3965 kB/s)                    
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
3 packages can be upgraded. Run 'apt list --upgradable' to see them.
```

### Actual Result

![image](https://github.com/user-attachments/assets/47b86cde-4c14-4a17-bc0d-1a28fa33526e)

**Comment by BlackDex** [comment]
Well, if it keeps saying that, it isn't able to access github.com and fetch content there.

So, either something blocks this request.

What do the Vaultwarden logs report?
Try setting `LOG_LEVEL=debug` if you do not see anything.

**Comment by BlackDex** [comment]
It might be the `HTTPS_PROXY` config is not valid for example.
Could you provide what you have configured for those values?

**Comment by BlackDex** [comment]
I just tested locally with `HTTPS_PROXY` both `socks5` and `socks5h`, and also regular `http` proxy. Both works as intended.

So either your `HTTPS_PROXY` is probably invalid or something on the proxy side is blocking stuff.

Please check the proxy logs.

**Comment by zmweske** [comment]
My `/admin/diagnostics` page looks very similar (but no internet proxy, only a reverse proxy) and I was able to wget the github page as used [here](https://github.com/dani-garcia/vaultwarden/blob/3baffeee9a167d61f3b7b71f8693b6cd4c5fdf43/src/api/admin.rs#L609), which appears to be how it checks for that `internet access` item. I am using the latest 1.33.2-alpine image and DNS appears to be working. Any thoughts?
![image](https://github.com/user-attachments/assets/b0f486ff-0021-47f2-9d5d-919beebd359a)

**Reply by zmweske** [reply]
I thought this was an issue interfering with the icon service, turns out it was an un-expiring icon miss cache and a very stale cache. 

I'm still curious what the problem above might be but impact is currently minimal, so it's a low priority

**Reply by BlackDex** [reply]
The looks should indicate what's going on there. 
For some reason Vaultwarden isn't able to contact GitHub and the time ntp web API.

So it's either blocked by a firewall, or something else in the docker network is causing an issue.

**Reply by zmweske** [reply]
The weird part is that I can hit github from inside the container and even download resources from it. I didn't test NTP directly but I know that it is synced with the host, which is working fine. I guess as of now it doesn't seem to have any effect with user experience but just weird nonetheless. Thanks for the reply!

**Reply by BlackDex** [reply]
So, what do the logs report then?

**Reply by zmweske** [reply]
I apologize, this slipped my mind but I pulled it up today and everything is green. I haven't even updated it- it's the same version so I'll try to report back if I see the error again.
